src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -326,23 +326,22 @@ class MavlinkStreamHeartbeat : public MavlinkStream
 
 	bool send(const hrt_abstime t)
 	{
-		struct vehicle_status_s status = {};
+		vehicle_status_s status;
 
 		/* always send the heartbeat, independent of the update status of the topics */
```

Comment:
This is not correct anymore. Please change it back to always send the heartbeat.

Reply:
I see now that the comment is wrong, but this is intentional. Commander publishes vehicle_status at >= 1 Hz. HEARTBEAT send will now sync up with commander publication and more importantly reflect if commander is actually alive.

Reply:
I see your reasoning, but I think it's more important to always send a heartbeat, even in the case commander is not running or misbehaving (or for some reasons the >1Hz assumption does not hold anymore). This makes link debugging and a lot easier.

Reply:
Fair enough, original HEARTBEAT restored.